Dockerfile

```diff
@@ -1,27 +1,89 @@
+RUN apt-get install -qq -y --no-install-recommends \
+    build-essential \
+    autoconf \
+    dh-autoreconf \
+    # @see https://help.ubuntu.com/community/CheckInstall
+    checkinstall \
+    curl \
+    # @see https://stedolan.github.io/jq/tutorial/
+    jq;
+# Install dev stuff for git
+RUN apt-get install -qq -y --no-install-recommends \
+    libcurl4-openssl-dev \
+    tcl-dev \
+    gettext \
+    asciidoc \
+    libexpat1-dev \
+    libz-dev \
+    libssl-dev;
+# prepare source dir
+RUN mkdir -p /root/src;
 WORKDIR /root/src
-COPY --from=git-compiler-helper \
-  /root/git-latest-url-finder.run \
-  /root/git-latest-url-finder.run
-RUN /root/git-latest-url-finder.run > git-tar-url.txt && \
-  echo "$(cat git-tar-url.txt)" && \
-  curl -L --retry 5 --url "$(cat git-tar-url.txt)" --output "git-source.tar.gz" && \
-  tar -xf "git-source.tar.gz" --strip 1
-RUN make configure
-RUN CFLAGS="-static" ./configure --with-openssl
-RUN make
-RUN make install
-ENTRYPOINT [ "git" ]
\ No newline at end of file
+# download & extract source tar
+RUN curl ${CURL_OPTIONS} \
+        --location \
+        --fail \
+        # assuming the top-most tag is the most recent version
+        $(curl ${CURL_OPTIONS} --silent --fail 'https://api.github.com/repos/git/git/tags' | jq '.[0].tarball_url' | tr -d '"') \
+    | tar xz --strip 1;
+# configure & compile git with OpenSSL
+RUN make configure; \
+    ./configure --prefix=/usr --with-openssl; \
+    make;
+# create debian package
+RUN checkinstall \
+    --default \
+    --type=debian \
+    --pkgname=git-openssl \
+    --pkgversion=$(./git --version | cut -d " " -f 3) \
+    --pkgarch=$(dpkg --print-architecture) \
+    --conflicts=git \
+    --requires="libcurl4,zlib1g \(\>= 1:1.2.0\),perl,libpcre3,libexpat1 \(\>= 2.0.1\),liberror-perl,libc6" \
+    --recommends="less,patch,ssh-client,ca-certificates" \
+    make install;
+# move created .deb file to a more scriptable filename
+RUN cp -v ./git-openssl_$(./git --version | cut -d " " -f 3)-1_$(dpkg --print-architecture).deb /root/git-openssl.deb;
+
+# install package on plain ubuntu
+FROM ubuntu:latest
+# copy package from temporary image
+COPY --from=git-compiler \
+    /root/git-openssl.deb \
+    /root/git-openssl.deb
+# install package dependencies
+RUN set -xe; \
+    apt-get update -qq -y; \
+    apt-get install -qq -y --no-install-recommends \
+        libcurl4 \
+        perl \
+        libexpat1 \
+        liberror-perl \
+        ca-certificates \
+        less \
+        patch \
+        ssh-client; \
+    apt-get clean; \
+    # install package
+    dpkg -i /root/git-openssl.deb; \
+    # create git config files
+    touch /root/.gitconfig /root/.git-credentials /etc/skel/.gitconfig /etc/skel/.git-credentials; \
+    # create 10 user- and group-IDs in case this image is run without privileges
+    bash -c 'for i in {1000..1010}; do groupadd -g ${i} -o "group${i}"; useradd -m -u ${i} -g ${i} -s /usr/bin/bash "user${i}"; done'
```

Comment:
Not a super big fan of creating `N` number of users before they're needed, but I've been bitten in the butt too many times by docker containers not being able to do non-root stuff to complain. (I've tried hedging in some stuff to add the user to the `/etc/password` file when the container comes alive in entrypoint scripts, but that still suffers chicken-and-egg problems with permissions, so I think this is the best naive solution to this problem I've seen in a while)